source/source_integration_test.go

```diff
@@ -97,16 +90,21 @@ func TestSource_Read_tableHasNoData(t *testing.T) {
 		is.NoErr(err)
 	}()
 
+	cancel()
+
 	src := NewSource()
 
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+	defer cancel()
+
 	err = src.Configure(ctx, cfg)
 	is.NoErr(err)
 
 	err = src.Open(ctx, nil)
 	is.NoErr(err)
 
 	_, err = src.Read(ctx)
-	is.Equal(err, sdk.ErrBackoffRetry)
+	is.Equal(err, context.DeadlineExceeded)
```

Comment:
We need to keep the previous error, because then Conduit will retry and try getting more records.